test/build-images.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+#
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+IMAGE_BUILDER_ARG=""
+if [ "$PROJECT" != "ml-pipeline-test" ]; then
+  COPIED_IMAGE_BUILDER_IMAGE=${GCR_IMAGE_BASE_DIR}/image-builder
+  echo "Copy image builder image to ${COPIED_IMAGE_BUILDER_IMAGE}"
+  yes | gcloud container images add-tag \
+    gcr.io/ml-pipeline-test/image-builder:v20181128-0.1.3-rc.1-109-ga5a14dc-e3b0c4 \
+    ${COPIED_IMAGE_BUILDER_IMAGE}:latest
+  IMAGE_BUILDER_ARG="-p image-builder-image=${COPIED_IMAGE_BUILDER_IMAGE}"
+fi
+
+# image caching can be turned on by setting $CACHE_IMAGES env flag to speed up
+# debugging
+BUILT_IMAGES=$(gcloud container images list --repository=${GCR_IMAGE_BASE_DIR})
```

Comment:
great change. I assume this is for speeding up /retest?

Reply:
It's initially meant for speeding up debugging tests locally.

I didn't enable it for /retest now (notice the env flag guard).

Do you think we can assume this is safe? This could cut 18min~ for /retest. Pretty useful if we can assume it doesn't change test results.

Reply:
The caching is based on commit hash, so I think this should be safe. Changed my mind, and I turned on caching by default.